haystack/preview/document_stores/protocols.py

```diff
@@ -51,69 +51,65 @@ def filter_documents(self, filters: Optional[Dict[str, Any]] = None) -> List[Doc
         """
         Returns the documents that match the filters provided.
 
-        Filters are defined as nested dictionaries. The keys of the dictionaries can be a logical operator (`"$and"`,
-        `"$or"`, `"$not"`), a comparison operator (`"$eq"`, `$ne`, `"$in"`, `$nin`, `"$gt"`, `"$gte"`, `"$lt"`,
-        `"$lte"`) or a metadata field name.
+        Filters are defined as nested dictionaries. There are two types of dictionaries:
+        - Comparison
+        - Logic
 
-        Logical operator keys take a dictionary of metadata field names and/or logical operators as value. Metadata
-        field names take a dictionary of comparison operators as value. Comparison operator keys take a single value or
-        (in case of `"$in"`) a list of values as value. If no logical operator is provided, `"$and"` is used as default
-        operation. If no comparison operator is provided, `"$eq"` (or `"$in"` if the comparison value is a list) is used
-        as default operation.
+        Top level must be either be a Logic dictionary.
```

Comment:
```suggestion
        The outer dictionary must be of type Logic.
```

Reply:
I actually would move this sentence after you show how the two types look

Reply:
Also this is wrong. 🤦 
I'll change it to "The outer dictionary must be of type Comparison or Logic."